superset/tags/models.py

```diff
@@ -168,19 +164,15 @@ def after_insert(
         connection: Connection,
         target: Dashboard | FavStar | Slice | Query | SqlaTable,
     ) -> None:
-        session = Session(bind=connection)
```

Comment:
The various SQLAlchemy hooks, `after_insert`, `after_delete`, etc. pass in the connection from which we instantiate a new session which was never closed. I believe we should be using the Flask-SQLAlchemy session here (and not committing) to ensure the DAO et al. logic is captured in a single transaction.